infrastructure/modules/observability/variables.tf

```diff
@@ -52,7 +52,7 @@ variable "app_insights_app_type" {
 
 variable "monitor_workspace_name" {
   type        = string
-  default     = ""
+  default     = null
   description = "Name for the Azure Monitor workspace."
```

Comment:
Is this description misleading now? I think it would be more correct to say that you specify this if you want to BYO resource. The script does not create the resource if this is set.